objects/src/accounts/package/storage_entry/mod.rs

```diff
@@ -0,0 +1,611 @@
+    pub fn new_multi_slot(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slots: Vec<u8>,
+        values: Vec<impl Into<WordRepresentation>>,
+    ) -> Result<Self, AccountComponentTemplateError> {
+        if slots.len() != values.len() {
+            return Err(AccountComponentTemplateError::InvalidMultiSlotEntry);
+        }
+
+        for window in slots.windows(2) {
+            if window[1] != window[0] + 1 {
+                return Err(AccountComponentTemplateError::NonContiguousSlots);
+            }
+        }
+
+        Ok(StorageEntry::MultiSlot {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slots,
+            values: values.into_iter().map(Into::into).collect(),
+        })
+    }
+
+    /// Returns the slot indices that the storage entry covers.
+    pub fn slot_indices(&self) -> &[u8] {
+        match self {
+            StorageEntry::MultiSlot { slots, .. } => slots.as_slice(),
+            StorageEntry::Value { slot, .. } => core::slice::from_ref(slot),
+            StorageEntry::Map { slot, .. } => core::slice::from_ref(slot),
+        }
+    }
+
+    /// Returns the name of the storage entry.
+    pub fn name(&self) -> &str {
+        match self {
+            StorageEntry::Value { name, .. } => name.as_str(),
+            StorageEntry::Map { name, .. } => name.as_str(),
+            StorageEntry::MultiSlot { name, .. } => name.as_str(),
+        }
+    }
+
+    /// Returns the optional description of the storage entry.
+    pub fn description(&self) -> Option<&str> {
+        match self {
+            StorageEntry::Value { description, .. } => description.as_deref(),
+            StorageEntry::Map { description, .. } => description.as_deref(),
+            StorageEntry::MultiSlot { description, .. } => description.as_deref(),
+        }
+    }
+
+    /// Returns all the `WordRepresentation` values covered by this entry.
+    /// For `Value` entries, this returns a single-element slice.
+    /// For `MultiSlot` entries, this returns all values.
+    /// For `Map` entries, since they're key-value pairs, return an empty slice.
+    pub fn word_values(&self) -> &[WordRepresentation] {
+        match self {
+            StorageEntry::Value { value, .. } => std::slice::from_ref(value),
+            StorageEntry::MultiSlot { values, .. } => values.as_slice(),
+            StorageEntry::Map { .. } => &[],
+        }
+    }
+
+    /// Returns the map entries for a `Map` variant as a slice.
+    /// Returns an empty slice for non-map variants.
+    pub fn map_entries(&self) -> &[MapEntry] {
+        match self {
+            StorageEntry::Map { values, .. } => values.as_slice(),
+            StorageEntry::Value { .. } => &[],
+            StorageEntry::MultiSlot { .. } => &[],
+        }
+    }
+
+    /// Returns an iterator over all of the storage entries's template keys.
+    // TODO: Should template keys be typed?
```

Comment:
Is this a leftover TODO?

Reply:
What I meant by this is that if `TemplateKey` should be typed. That is, whether it should express what type of value it expects. That is currently not the case (all that it expresses by itself is the name of the key).

Reply:
Ah, I thought that `TemplateKey` was perhaps a later introduction and it was previously a `String`. 

I like the idea of typing it. It would definitely catch more errors. So right now this would be only word and felt, right, so maybe something like `"{{foo.key: Word}}"` or `"{{foo.bar: Felt}}"`?
I guess this could be done in a follow-up PR.